pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -326,10 +328,50 @@ func (pl *VolumeRestrictions) EventsToRegister() []framework.ClusterEventWithHin
 		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add}},
 		// Pods may fail to schedule because the PVC it uses has not yet been created.
 		// This PVC is required to exist to check its access modes.
-		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add | framework.Update},
+			QueueingHintFn: pl.isSchedulableAfterPersistentVolumeClaimChange},
 	}
 }
 
+// isSchedulableAfterPersistentVolumeClaimChange is invoked whenever a PersistentVolumeClaim added or changed, It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *VolumeRestrictions) isSchedulableAfterPersistentVolumeClaimChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	oldPersistentVolumeClaim, newPersistentVolumeClaim, err := util.As[*v1.PersistentVolumeClaim](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, fmt.Errorf("unexpected objects in isSchedulableAfterPersistentVolumeClaimChange: %w", err)
+	}
+
+	if oldPersistentVolumeClaim != nil || newPersistentVolumeClaim.Namespace != pod.Namespace {
+		return framework.QueueSkip, nil
+	}
+
+	pvcs := sets.New[string]()
+
+	for _, volume := range pod.Spec.Volumes {
+		if volume.PersistentVolumeClaim == nil {
+			continue
+		}
+
+		pvc, err := pl.pvcLister.PersistentVolumeClaims(pod.Namespace).Get(volume.PersistentVolumeClaim.ClaimName)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				logger.V(5).Info("The PVC for the Pod is not created, and this Pod won't be schedulable until the PVC is created",
+					"pod", klog.KObj(pod), "pvc", volume.PersistentVolumeClaim.ClaimName, "err", err)
+				return framework.QueueSkip, nil
+			}
+			return framework.Queue, err
+		}
+
+		pvcs.Insert(pvc.Name)
+	}
+
+	// We're only interested in PVC which the Pod requests.
+	if oldPersistentVolumeClaim == nil && pvcs.Has(newPersistentVolumeClaim.Name) {
+		return framework.Queue, nil
+	}
+	return framework.QueueSkip, nil
+}
```

Comment:
Because of the throughput concern of QHint, we don't want to refer to other resources via listerner (= `pl.pvcLister.PersistentVolumeClaims().Get()` in this PR) as a starting point _for now_. 
(We may change this direction after we establish the observability around QHints: https://github.com/kubernetes/kubernetes/issues/124566)

So, let's make one step simpler:

```suggestion
		if volume.PersistentVolumeClaim.ClaimName == newPersistentVolumeClaim.Name {
			logger.V(5).Info("PVC that is referred from the pod was created, which might make this pod schedulable", "pod", klog.KObj(pod), "PVC", klog.KObj(newPersistentVolumeClaim))
			return framework.Queue, nil
		}
	}

	logger.V(5).Info("PVC irrelevant to the Pod was created, which doesn't make this pod schedulable", "pod", klog.KObj(pod), "PVC", klog.KObj(newPersistentVolumeClaim))
	return framework.QueueSkip, nil
}
```

This could eliminate `pl.pvcLister.PersistentVolumeClaims().Get()`, while we keep a filtering to some extent.

Reply:
fixed.